File: 2025/java/src/test/java/nl/dirkgroot/adventofcode/year2025/Day00Test.java
package nl.dirkgroot.adventofcode.year2025;

import nl.dirkgroot.adventofcode.util.AoCTest;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class Day00Test implements AoCTest {
    @Test
    public void part1Solution() {
        assertThat(invokeWith(this::part1, input(0))).isEqualTo(26);
    }

    @Test
    public void part2Solution() {
        assertThat(invokeWith(this::part2, input(0))).isEqualTo(52);
    }

    long part1(String input) {
        return input.length();
    }

    long part2(String input) {
        return input.length() * 2L;
    }
}

File: 2024/kotlin/src/test/kotlin/nl/dirkgroot/adventofcode/Day00Test.kt
package nl.dirkgroot.adventofcode

import assertk.assertThat
import assertk.assertions.isEqualTo
import nl.dirkgroot.adventofcode.util.input
import nl.dirkgroot.adventofcode.util.invokedWith
import org.junit.jupiter.api.Test

class Day00Test {
    @Test
    fun `part1 solution`() {
        assertThat(::part1 invokedWith input(0)).isEqualTo(21)
    }

    fun part1(input: String) = input.length
}
